pinot-common/src/main/java/org/apache/pinot/pql/parsers/pql2/ast/FilterKind.java

```diff
@@ -48,4 +48,13 @@ public boolean isRange() {
     return this == GREATER_THAN || this == GREATER_THAN_OR_EQUAL || this == LESS_THAN || this == LESS_THAN_OR_EQUAL
         || this == BETWEEN || this == RANGE;
   }
+
+  /**
+   * Returns true if the filter is a predicate. Returns false otherwise. This logic should mimic FilterContext.Type.
+   *
+   * @return where the filter is a predicate.
+   */
+  public boolean isPredicate() {
+    return this != AND && this != OR && this != NOT;
+  }
```

Comment:
This, along with the method `isRange` above it, is a little messy - it would be better to add fields to the enum `_isRange` and `_isPredicate` and set them in the enum initialiser.

Reply:
Done.